src/OrleansAWSUtils/Storage/SQSStorage.cs

```diff
@@ -0,0 +1,233 @@
+            {
+                var value = serviceConfig.Split(new[] { '=' }, StringSplitOptions.RemoveEmptyEntries);
+                if (value.Length == 2 && !string.IsNullOrWhiteSpace(value[1]))
+                    service = value[1];
+            }
+
+            var secretKeyConfig = parameters.Where(p => p.Contains(SecretKeyPropertyName)).FirstOrDefault();
+            if (!string.IsNullOrWhiteSpace(secretKeyConfig))
+            {
+                var value = secretKeyConfig.Split(new[] { '=' }, StringSplitOptions.RemoveEmptyEntries);
+                if (value.Length == 2 && !string.IsNullOrWhiteSpace(value[1]))
+                    secretKey = value[1];
+            }
+
+            var accessKeyConfig = parameters.Where(p => p.Contains(AccessKeyPropertyName)).FirstOrDefault();
+            if (!string.IsNullOrWhiteSpace(accessKeyConfig))
+            {
+                var value = accessKeyConfig.Split(new[] { '=' }, StringSplitOptions.RemoveEmptyEntries);
+                if (value.Length == 2 && !string.IsNullOrWhiteSpace(value[1]))
+                    accessKey = value[1];
+            }
+        }
+
+        private void CreateClient()
+        {
+            var credentials = new BasicAWSCredentials(accessKey, secretKey);
+            sqsClient = new AmazonSQSClient(credentials, new AmazonSQSConfig { RegionEndpoint = AWSUtils.GetRegionEndpoint(service) });
+        }
+
+        private async Task<string> GetQueueUrl()
+        {
+            try
+            {
+                var response = await sqsClient.GetQueueUrlAsync(QueueName);
+                if (!string.IsNullOrWhiteSpace(response.QueueUrl))
+                    queueUrl = response.QueueUrl;
+
+                return queueUrl;
+            }
+            catch (QueueDoesNotExistException)
+            {
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Initialize SQSStorage by creating or connecting to an existent queue
+        /// </summary>
+        /// <returns></returns>
+        public async Task InitQueueAsync()
+        {
+            try
+            {
+                if (string.IsNullOrWhiteSpace(await GetQueueUrl()))
+                {
+                    var response = await sqsClient.CreateQueueAsync(QueueName);
+                    queueUrl = response.QueueUrl;
+                }
+            }
+            catch (Exception exc)
+            {
+                ReportErrorAndRethrow(exc, "InitQueueAsync", ErrorCode.StreamProviderManagerBase);
+            }
+        }
+
+        /// <summary>
+        /// Delete the queue
+        /// </summary>
+        /// <returns></returns>
+        public Task DeleteQueue()
+        {
+            try
+            {
+                if (string.IsNullOrWhiteSpace(queueUrl))
+                    throw new InvalidOperationException("Queue not initialized");
+                return sqsClient.DeleteQueueAsync(queueUrl);
```

Comment:
await instead of return.  By returning delete, exceptions will not be caught in below catch.